bank_account_project.py

```diff
@@ -1,169 +0,0 @@
-'''
-***All credit for the data sourced for this project goes to exchangerates.org.uk***
-
-This program simulates a transaction with an ATM machine. If the user decides to 
-withdraw money from their bank account, they are given the option to enter the 
-withdrawal amount in a foreign currency. The program will grab the most updated 
-currency exchange figures and return the amount remaining in the bank account 
-in GBP. 
-'''
-
-#This function scrapes conversion rates between GBP(£) and some of the most 
-#common currencies in the world. The function takes one argument, gbp, that 
-#must be either an integar or a float. 
-def currency_func(gbp):
-
-    import requests
-    from bs4 import BeautifulSoup
-    
-    URL = 'https://www.exchangerates.org.uk/'
-    page = requests.get(URL,'html.parser')
-    soup = BeautifulSoup(page.content, 'html.parser')
-    
-    #grabbing the desired div and table for which the desired data is housed
-    table = soup.find('table')
-    
-    #grabbing all of the desired table data 
-    eur_exch = float(table.find_all('td')[3].text)
-    usd_exch = float(table.find_all('td')[8].text)
-    nzd_exch = float(table.find_all('td')[13].text)
-    aud_exch = float(table.find_all('td')[18].text)
-    cad_exch = float(table.find_all('td')[23].text)
-    jpy_exch = float(table.find_all('td')[28].text)
-    zar_exch = float(table.find_all('td')[33].text)
-    aed_exch = float(table.find_all('td')[38].text)
-    inr_exch = float(table.find_all('td')[43].text)
-    try_exch = float(table.find_all('td')[48].text)
-    chf_exch = float(table.find_all('td')[53].text)
-    
-    exchange_rates = [eur_exch, usd_exch, nzd_exch, aud_exch, cad_exch,
-                      jpy_exch, zar_exch, aed_exch, inr_exch, try_exch,
-                      chf_exch]
-    
-    return [gbp * x for x in exchange_rates]
-
-
-class BankAccount(object):
-
-    
-    def __init__(self,balance=0.0):
-        self.balance = balance
-
-    
-    def withdraw(self):
-        currency = str(input('''You may withdraw your money in the following currencies: 
-        GBP, EUR, USD, NZD, AUD, CAD, JPY, ZAR, AED, INR, TRY, CHF. 
-        What currency do you want to withdraw your money in? '''))
-        
-        # Should implement error handling here for when an invalid currency is entered.
-        
-        while True:
-           withdraw_amount = float(input('How many ' + currency.upper() + 's would you like to withdraw? '))
-           
-           # Each if statement calculates the withdrawal amount in GBP for any of the given currencies
-           # Should try to make this portion of the program more elegent with a function
-           
-           if currency.upper() == 'EUR':
-               withdraw_amount /= float(currency_func(1)[0])
```

Comment:
So again here - rather than checking each currency with a hard coded index, you can look in the dict for the currency:

```
withdraw_amt /=  currency_dict[currency.upper()]
```